dspace/modules/journal-submit/src/main/java/org/datadryad/submission/DryadEmailSubmission.java

```diff
@@ -535,14 +548,46 @@ public String getServletInfo() {
 
     /**
      * Returns a PrintWriter with the correct character encoding set.
-     * 
+     *
      * @param aResponse In which to set the character encoding
      * @return A <code>PrintWriter</code> to send text through
      * @throws IOException If there is trouble getting a writer
      */
     private PrintWriter getWriter(HttpServletResponse aResponse)
-    throws IOException {
+            throws IOException {
         aResponse.setContentType("xml/application; charset=UTF-8");
         return aResponse.getWriter();
     }
+
+    private class DryadEmailSubmissionHarvester extends TimerTask {
+        @Override
+        public void run() {
+            ArrayList<String> labels = new ArrayList<String>();
+            labels.add("journal-submit");
+            try {
+                List<Message> messages = dryadGmailService.retrieveMessagesWithLabels(labels);
+                // Print ID of each Thread.
+                if (messages != null) {
+                    ArrayList<String> processedMessageIDs = new ArrayList<String>();
+                    LOGGER.info("got " + messages.size() + " messages");
+                    for (Message message : messages) {
+                        LOGGER.info("Message: "+ message.getId() + ", " + message.getSnippet());
+                        ByteArrayInputStream postBody = new java.io.ByteArrayInputStream(Base64.decodeBase64(message.getRaw()));
```

Comment:
Since you've already encapsulated the Gmail communication in DryadGmailService, it may be a good idea to encapsulate the messages in a class too - and handle the base64 decoding and other Gmail implementation-specific details there.